### PR TITLE
[5.0] Added remaining Filesystem tests

### PR DESCRIPTION
**Pull request by manan-jadhav**
I've removed the `testGetFiles()` because the same logic is implemented in two different test cases, `testDirectoriesFindsDirectories()` and `testFilesMethod()`.

### CODE REVIEW

**Comment by CurosMJ** [comment]
Follow on for #6893

File: tests/Filesystem/FilesystemTest.php [hunk 1]
```diff
@@ -137,32 +137,163 @@ public function testCopyDirectoryMovesEntireDirectory()
...
+	}
+
+
+	public function testDirectoriesFindsDirectories()
+	{
+		mkdir(__DIR__.'/foo');
+		mkdir(__DIR__.'/bar');
+		$files = new Filesystem;
+		$directories = $files->directories(__DIR__);
+		$this->assertContains(__DIR__.DIRECTORY_SEPARATOR.'foo',$directories);
```
Comment (lucasmichot) [inline tests/Filesystem/FilesystemTest.php]: Use / instead of DIRECTORY_SEPARATOR

File: tests/Filesystem/FilesystemTest.php [hunk 2]
```diff
@@ -137,32 +137,163 @@ public function testCopyDirectoryMovesEntireDirectory()
...
+	}
+
+
+	public function testAllFilesFindsFiles()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		file_put_contents(__DIR__.'/bar.txt', 'bar');
+		$files = new Filesystem;
+		$allFiles = [];
+		foreach($files->allFiles(__DIR__) as $file)
```
Comment (lucasmichot) [inline tests/Filesystem/FilesystemTest.php]: missing curly brackets

File: tests/Filesystem/FilesystemTest.php (hunk 1 above)
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: @lucasmichot actually you're wrong. Run the tests on windows and see what happens.

Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: there is a missing space after the comma though

File: tests/Filesystem/FilesystemTest.php (hunk 2 above)
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: and missing space

File: tests/Filesystem/FilesystemTest.php (hunk 1 above)
Comment (lucasmichot) [inline tests/Filesystem/FilesystemTest.php]: Strange `FileStore::put` can work

Comment (lucasmichot) [inline tests/Filesystem/FilesystemTest.php]: I mean `path`

Comment (CurosMJ) [inline tests/Filesystem/FilesystemTest.php]: @lucasmichot When we get paths from the system, for example by `glob()`, the directory separator used is according to the system, for Windows, it's a `\`, so it's best to use `DIRECTORY_SEPARATOR`, when we are equating the paths as strings.

**Pull request merged**